Improve type hints

outdooracorn opened the pull request:
Bug: T343886

outdooracorn committed:
Improve type hints

Bug: T343886

outdooracorn commented on pagesections/ComponentInteraction.js:
```diff
@@ -15,18 +15,30 @@ const ComponentInteraction = ( Base ) => class extends Base {
+	 */
 	setValueOnLookupElement( element, value ) {
 		element.$( 'input' ).setValue( value );
 		$( 'body' ).$( this.constructor.OOUI_SELECTORS.LOOKUP_OPTION_WIDGET ).waitForDisplayed();
```
Any way to have these `OOUI_SELECTORS` detected by the IDE? They are detected correctly in `item.page.js`.

jakobw replied:
Giving the classes a name seems to be enough to help the IDE.

outdooracorn replied:
Yes, that works, thanks! \o/

jakobw reviewed (dismissed):
Yay, typehints! :rocket: 
Up to you whether you want to try to improve try out fixing the static getters here or in a separate PR. Feel free to merge.

outdooracorn committed:
Simplify `waitUntil()` in `main.statement.section.js`

Similar to https://github.com/webdriverio/example-recipes/blob/2cd4ba00922813ad8e55e297a5c7c230dba62f86/waitUntil/waitUntilExample.js

Bug: T343886

jakobw commented on pageobjects/item.page.js:
```diff
@@ -7,6 +7,11 @@ const Page = require( 'wdio-mediawiki/Page' ),
 	PageMixture = MixinBuilder.mix( Page ).with( MainStatementSection, ComponentInteraction, TaintedRefSection );
 
+/**
+ * @mixes MainStatementSectionMixin
+ * @mixes ComponentInteractionMixin
+ * @mixes TaintedRefSectionMixin
```
Ooh, TIL about `@mixes`. Neat!

jakobw commented on pagesections/ComponentInteraction.js:
```diff
@@ -1,47 +1,63 @@
-		await $( this.constructor.OOUI_SELECTORS.VISIBLE_ENTITY_SUGGESTION ).click();
-	}
+const ComponentInteraction = ( Base ) => {
+	/** @extends Base */
+	return class ComponentInteractionMixin extends Base {
```
Hmm, what's the reason for this change? `/** @extends Base */` is pretty redundant to begin with since that's what the line below says, and since `Base` is dynamic, there are no type hints to be unlocked anyway. If there is no benefit I'm missing, I'd prefer getting rid of this one and going back to the shorthand return syntax to remove the extra indentation level.

Same in the two other cases below.

outdooracorn replied:
Good question! It does indeed look to be redundant, but I found that it was required in order for the IDE to realize that `ItemPage` has methods from the "pagesections" classes (e.g. `ItemPage.selectFirstSuggestedEntityOnEntitySelector()`). If you can think of a better way to hint that I'm all ears.

jakobw replied:
but... why...

jakobw reviewed (approved):
I don't understand why the `@extends` annotation does what it does here, but it works, so hurray! :tada:
